angular-ngc/defs.bzl

```diff
@@ -238,7 +244,7 @@ def ng_library(name, package_name = None, deps = [], test_deps = [], visibility
       visibility: visibility of the primary targets ({name}, 'test')
     """
 
-    test_spec_srcs = native.glob(["src/**/*.spec.ts"])
+    test_spec_srcs = native.glob(["src/**/*.spec.ts", "src/test.ts"])
```

Comment:
The test bootstrapping needs to be done before the tests so we can't just include it as a spec file. It need to be included when we run esbuild so it shares code with the specs (such as angular), but it needs to be in a `<script>` before the spec files.

Reply:
Got it, but in the `rules_nodejs` sample here https://github.com/bazelbuild/rules_nodejs/blob/stable/examples/angular/tools/karma.conf.js#L201-L206, they are included as normal `config.files`, only they are loaded before the normal `spec files`.

Reply:
Yeah that makes sense, but we need to distinguish them from the specs so I don't think we can put them into this array like this...?

Reply:
I also use `write_file` to auto generate this `test_setup.ts` and move it into `unit_test` macro.